src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,139 @@
+
+  const dispatch = useDispatch();
+
+  const todos = useSelector(getTodosSelector);
+
+  useEffect(() => {
+    const loadTodosFromServer = async () => {
+      const todosFromServer = await getTodos();
+
+      dispatch(LoadTodosAction(todosFromServer));
+    };
+
+    loadTodosFromServer();
+  }, []);
+
+  const selectUser = async (userId: number) => {
+    if (selectedUserId !== userId) {
+      const userFromServer = await getUser(userId);
+
+      setSelectedUserId(userId);
+
+      dispatch(LoadUserAction(userFromServer));
+    }
+  };
+
+  let preparedTodos = [...todos];
+
+  useMemo(() => {
+    if (searchQuery) {
+      preparedTodos = [...todos].filter(todo => todo.title.includes(searchQuery));
+    }
+
+    if (filterBy !== 'nofilter') {
+      preparedTodos = filterBy === 'completed'
+        ? preparedTodos.filter(todo => todo.completed)
+        : preparedTodos.filter(todo => !todo.completed);
+    }
+  }, [searchQuery, filterBy]);
+
+  const handleSearchInput = (input: string) => {
+    setSearchQuery(input);
+  };
+
+  const removeTodo = async (todoId: number) => {
+    const todosFromServer = await deleteTodo(todoId);
+
+    dispatch(LoadTodosAction(todosFromServer));
+  };
+
+  return (
+    <div className="TodoList">
+      <h2>Todos:</h2>
+
+      <div className="TodoList__search">
+        <input
+          type="text"
+          placeholder="Search..."
+          className="TodoList__searchbar"
+          value={searchQuery}
+          onChange={event => handleSearchInput(event.target.value)}
+        />
+        <select
+          name=""
+          id=""
+          className="TodoList__selector"
+          value={filterBy}
+          onChange={event => setFilterBy(event.target.value)}
+        >
+          <option value="nofilter">all</option>
+          <option value="active">active</option>
+          <option value="completed">completed</option>
+        </select>
+      </div>
+
+      <div className="TodoList__list-container">
+        <ul className="TodoList__list">
+          {preparedTodos.map(todo => (
+            <li
+              key={todo.id}
+              className={classNames(
+                'TodoList__item',
+                { 'TodoList__item--checked': todo.completed },
+                { 'TodoList__item--unchecked': !todo.completed },
+              )}
+            >
+              <label htmlFor="isCompleted">
+                <input
+                  id="isCompleted"
+                  type="checkbox"
+                  checked={todo.completed}
+                  readOnly
+                />
+                <p>{todo.title}</p>
+              </label>
+
+              <div className="TodoList__button-container">
+                <button
+                  className={classNames(
+                    'button',
+                    'TodoList__user-button',
+                    { 'TodoList__user-button--selected': todo.userId === selectedUserId },
```

Comment:
selectedUserId is not needed, you have selected user in redux store use him here
```suggestion
                    { 'TodoList__user-button--selected': todo.userId ===user.id },
```